==29752== Memcheck, a memory error detector
==29752== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==29752== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==29752== Command: /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/compiler-singlepass
==29752== 
==29752== 
==29752== HEAP SUMMARY:
==29752==     in use at exit: 30,336 bytes in 90 blocks
==29752==   total heap usage: 366 allocs, 276 frees, 93,648 bytes allocated
==29752== 
==29752== 8 bytes in 1 blocks are still reachable in loss record 1 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA523C2: new<alloc::sync::Arc<rayon_core::registry::Registry>> (boxed.rs:177)
==29752==    by 0xA523C2: rayon_core::util::leak (util.rs:5)
==29752==    by 0xA576BE: rayon_core::registry::set_global_registry::{{closure}}::{{closure}} (registry.rs:196)
==29752==    by 0xA8D6F8: core::result::Result<T,E>::map (result.rs:519)
==29752==    by 0xA57759: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xA7AB90: std::sync::once::Once::call_once (once.rs:264)
==29752==    by 0xA57666: rayon_core::registry::set_global_registry (registry.rs:194)
==29752== 
==29752== 32 bytes in 1 blocks are still reachable in loss record 2 of 30
==29752==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==29752==    by 0x4E3F165: dlsym (dlsym.c:70)
==29752==    by 0x1185853: fetch (weak.rs:61)
==29752==    by 0x1185853: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==29752==    by 0x1185853: min_stack_size (thread.rs:449)
==29752==    by 0x1185853: std::sys::unix::thread::Thread::new (thread.rs:49)
==29752==    by 0xA870FB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752== 
==29752== 64 bytes in 4 blocks are still reachable in loss record 3 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA660A6: new<crossbeam_deque::deque::Buffer<rayon_core::job::JobRef>> (boxed.rs:177)
==29752==    by 0xA660A6: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==29752==    by 0xA82953: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==29752==    by 0xA82AA3: crossbeam_epoch::atomic::Atomic<T>::init (atomic.rs:309)
==29752==    by 0xA82A73: crossbeam_epoch::atomic::Atomic<T>::new (atomic.rs:294)
==29752==    by 0xA647A6: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:253)
==29752==    by 0xA586AE: rayon_core::registry::Registry::new::{{closure}} (registry.rs:229)
==29752==    by 0xA913BC: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA6200D: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752== 
==29752== 64 bytes in 4 blocks are still reachable in loss record 4 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x11857AC: alloc (alloc.rs:80)
==29752==    by 0x11857AC: alloc (alloc.rs:174)
==29752==    by 0x11857AC: exchange_malloc (alloc.rs:277)
==29752==    by 0x11857AC: std::sys::unix::thread::Thread::new (thread.rs:44)
==29752==    by 0xA870FB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xA7AB90: std::sync::once::Once::call_once (once.rs:264)
==29752==    by 0xA57666: rayon_core::registry::set_global_registry (registry.rs:194)
==29752== 
==29752== 72 bytes in 1 blocks are still reachable in loss record 5 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xD7779B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xD778C3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xD776FC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xD7A5B7: std::sync::rwlock::RwLock<T>::new (rwlock.rs:134)
==29752==    by 0xD7A8EC: <std::sync::rwlock::RwLock<T> as core::default::Default>::default (rwlock.rs:451)
==29752==    by 0xD6C260: __static_ref_initialize (frame_info.rs:28)
==29752==    by 0xD6C260: core::ops::function::FnOnce::call_once (function.rs:233)
==29752==    by 0xDA3F7B: lazy_static::lazy::Lazy<T>::get::{{closure}} (inline_lazy.rs:31)
==29752==    by 0xDA28F4: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xDA2870: std::sync::once::Once::call_once (once.rs:264)
==29752==    by 0xD67688: get<std::sync::rwlock::RwLock<wasmer_engine::trap::frame_info::GlobalFrameInfo>,fn() -> std::sync::rwlock::RwLock<wasmer_engine::trap::frame_info::GlobalFrameInfo>> (inline_lazy.rs:30)
==29752==    by 0xD67688: __stability (lib.rs:142)
==29752==    by 0xD67688: <wasmer_engine::trap::frame_info::FRAME_INFO as core::ops::deref::Deref>::deref (lib.rs:144)
==29752== 
==29752== 128 bytes in 4 blocks are still reachable in loss record 6 of 30
==29752==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==29752==    by 0x1151DDE: register_dtor (fast_thread_local.rs:33)
==29752==    by 0x1151DDE: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==29752==    by 0x1151DDE: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==29752==    by 0x11745D9: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==29752==    by 0x11745D9: __getit (local.rs:175)
==29752==    by 0x11745D9: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==29752==    by 0x11745D9: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==29752==    by 0x11745D9: std::sys_common::thread_info::set (thread_info.rs:40)
==29752==    by 0xA874C1: std::thread::Builder::spawn_unchecked::{{closure}} (mod.rs:473)
==29752==    by 0xA7BBDE: core::ops::function::FnOnce::call_once{{vtable-shim}} (function.rs:233)
==29752==    by 0x1185BC9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==29752==    by 0x1185BC9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==29752==    by 0x1185BC9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==29752==    by 0x52516DA: start_thread (pthread_create.c:463)
==29752==    by 0x57A271E: clone (clone.S:95)
==29752== 
==29752== 128 bytes in 4 blocks are still reachable in loss record 7 of 30
==29752==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==29752==    by 0xABF431: std::thread::local::fast::Key<T>::try_register_dtor (local.rs:438)
==29752==    by 0xABF332: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==29752==    by 0xABF4D6: std::thread::local::fast::Key<T>::get (local.rs:410)
==29752==    by 0xABF22C: crossbeam_epoch::default::HANDLE::__getit (local.rs:175)
==29752==    by 0xA68ED4: std::thread::local::LocalKey<T>::try_with (local.rs:262)
==29752==    by 0xA6E862: crossbeam_epoch::default::with_handle (default.rs:43)
==29752==    by 0xA6EB46: crossbeam_epoch::default::is_pinned (default.rs:30)
==29752==    by 0xA6496C: crossbeam_deque::deque::Stealer<T>::steal (deque.rs:619)
==29752==    by 0xA5A512: rayon_core::registry::WorkerThread::steal::{{closure}} (registry.rs:779)
==29752==    by 0xA558FC: core::iter::traits::iterator::Iterator::find_map::check::{{closure}} (iterator.rs:2258)
==29752== 
==29752== 160 bytes in 4 blocks are still reachable in loss record 8 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA8580F: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29752==    by 0xA99020: rayon_core::latch::LockLatch::new (latch.rs:230)
==29752==    by 0xA5945C: rayon_core::registry::ThreadInfo::new (registry.rs:583)
==29752==    by 0xA7BAF1: core::ops::function::FnMut::call_mut (function.rs:154)
==29752==    by 0xA910D3: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA74A73: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92988: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752==    by 0xA90889: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29752== 
==29752== 160 bytes in 4 blocks are still reachable in loss record 9 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA8580F: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29752==    by 0xA99020: rayon_core::latch::LockLatch::new (latch.rs:230)
==29752==    by 0xA59474: rayon_core::registry::ThreadInfo::new (registry.rs:584)
==29752==    by 0xA7BAF1: core::ops::function::FnMut::call_mut (function.rs:154)
==29752==    by 0xA910D3: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA74A73: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92988: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752==    by 0xA90889: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29752== 
==29752== 160 bytes in 4 blocks are still reachable in loss record 10 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA8580F: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==29752==    by 0xA89183: <std::sync::mutex::Mutex<T> as core::default::Default>::default (mutex.rs:446)
==29752==    by 0xA73E6E: <rayon_core::sleep::WorkerSleepState as core::default::Default>::default (mod.rs:55)
==29752==    by 0xA7129D: <crossbeam_utils::cache_padded::CachePadded<T> as core::default::Default>::default (cache_padded.rs:77)
==29752==    by 0xA7270A: rayon_core::sleep::Sleep::new::{{closure}} (mod.rs:67)
==29752==    by 0xA91280: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA61E18: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92A59: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752== 
==29752== 160 bytes in 4 blocks are still reachable in loss record 11 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x115317E: alloc (alloc.rs:80)
==29752==    by 0x115317E: alloc (alloc.rs:174)
==29752==    by 0x115317E: exchange_malloc (alloc.rs:277)
==29752==    by 0x115317E: new<()> (mutex.rs:223)
==29752==    by 0x115317E: std::thread::Thread::new (mod.rs:1149)
==29752==    by 0xA86E9D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xA7AB90: std::sync::once::Once::call_once (once.rs:264)
==29752==    by 0xA57666: rayon_core::registry::set_global_registry (registry.rs:194)
==29752== 
==29752== 160 bytes in 4 blocks are still reachable in loss record 12 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA50867: alloc::sync::Arc<T>::new (sync.rs:324)
==29752==    by 0xA86F3B: std::thread::Builder::spawn_unchecked (mod.rs:465)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752== 
==29752== 192 bytes in 4 blocks are still reachable in loss record 13 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1170276: alloc (alloc.rs:80)
==29752==    by 0x1170276: alloc (alloc.rs:174)
==29752==    by 0x1170276: exchange_malloc (alloc.rs:277)
==29752==    by 0x1170276: std::sync::condvar::Condvar::new (condvar.rs:130)
==29752==    by 0xA9903C: rayon_core::latch::LockLatch::new (latch.rs:231)
==29752==    by 0xA5945C: rayon_core::registry::ThreadInfo::new (registry.rs:583)
==29752==    by 0xA7BAF1: core::ops::function::FnMut::call_mut (function.rs:154)
==29752==    by 0xA910D3: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA74A73: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92988: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752==    by 0xA90889: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29752==    by 0xA78461: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==29752==    by 0xA790E4: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29752==    by 0xA7A086: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29752== 
==29752== 192 bytes in 4 blocks are still reachable in loss record 14 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1170276: alloc (alloc.rs:80)
==29752==    by 0x1170276: alloc (alloc.rs:174)
==29752==    by 0x1170276: exchange_malloc (alloc.rs:277)
==29752==    by 0x1170276: std::sync::condvar::Condvar::new (condvar.rs:130)
==29752==    by 0xA9903C: rayon_core::latch::LockLatch::new (latch.rs:231)
==29752==    by 0xA59474: rayon_core::registry::ThreadInfo::new (registry.rs:584)
==29752==    by 0xA7BAF1: core::ops::function::FnMut::call_mut (function.rs:154)
==29752==    by 0xA910D3: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA74A73: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92988: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752==    by 0xA90889: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29752==    by 0xA78461: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==29752==    by 0xA790E4: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29752==    by 0xA7A086: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29752== 
==29752== 192 bytes in 4 blocks are still reachable in loss record 15 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1170276: alloc (alloc.rs:80)
==29752==    by 0x1170276: alloc (alloc.rs:174)
==29752==    by 0x1170276: exchange_malloc (alloc.rs:277)
==29752==    by 0x1170276: std::sync::condvar::Condvar::new (condvar.rs:130)
==29752==    by 0xA73E8A: <rayon_core::sleep::WorkerSleepState as core::default::Default>::default (mod.rs:57)
==29752==    by 0xA7129D: <crossbeam_utils::cache_padded::CachePadded<T> as core::default::Default>::default (cache_padded.rs:77)
==29752==    by 0xA7270A: rayon_core::sleep::Sleep::new::{{closure}} (mod.rs:67)
==29752==    by 0xA91280: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA61E18: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92A59: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752==    by 0xA906BA: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==29752==    by 0xA78827: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==29752==    by 0xA79041: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29752==    by 0xA7A025: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29752== 
==29752== 192 bytes in 4 blocks are still reachable in loss record 16 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x11531E2: alloc (alloc.rs:80)
==29752==    by 0x11531E2: alloc (alloc.rs:174)
==29752==    by 0x11531E2: exchange_malloc (alloc.rs:277)
==29752==    by 0x11531E2: new (condvar.rs:130)
==29752==    by 0x11531E2: std::thread::Thread::new (mod.rs:1150)
==29752==    by 0xA86E9D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xA7AB90: std::sync::once::Once::call_once (once.rs:264)
==29752==    by 0xA57666: rayon_core::registry::set_global_registry (registry.rs:194)
==29752== 
==29752== 320 bytes in 4 blocks are still reachable in loss record 17 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1153231: alloc (alloc.rs:80)
==29752==    by 0x1153231: alloc (alloc.rs:174)
==29752==    by 0x1153231: exchange_malloc (alloc.rs:277)
==29752==    by 0x1153231: new<std::thread::Inner> (sync.rs:324)
==29752==    by 0x1153231: std::thread::Thread::new (mod.rs:1145)
==29752==    by 0xA86E9D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xA7AB90: std::sync::once::Once::call_once (once.rs:264)
==29752==    by 0xA57666: rayon_core::registry::set_global_registry (registry.rs:194)
==29752== 
==29752== 384 bytes in 1 blocks are still reachable in loss record 18 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xAE7D9B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xAE8643: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xADF15A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29752==    by 0xA9BDE0: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29752==    by 0xA9A733: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29752==    by 0xA9E332: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29752==    by 0xA77218: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29752==    by 0xA78377: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2136)
==29752==    by 0xA790E4: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29752==    by 0xA7A086: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==29752==    by 0xA904E8: core::iter::traits::iterator::Iterator::collect (iterator.rs:1671)
==29752== 
==29752== 416 bytes in 4 blocks are still reachable in loss record 19 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA87067: new<closure-0> (boxed.rs:177)
==29752==    by 0xA87067: std::thread::Builder::spawn_unchecked (mod.rs:494)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752== 
==29752== 512 bytes in 1 blocks are still reachable in loss record 20 of 30
==29752==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1175CE9: aligned_malloc (alloc.rs:95)
==29752==    by 0x1175CE9: alloc (alloc.rs:22)
==29752==    by 0x1175CE9: __rdl_alloc (alloc.rs:304)
==29752==    by 0xAE7D9B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xAE8643: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xADF15A: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==29752==    by 0xA9C140: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==29752==    by 0xA9A633: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==29752==    by 0xA9E292: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==29752==    by 0xA771B8: alloc::vec::Vec<T>::reserve (vec.rs:505)
==29752==    by 0xA78767: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2136)
==29752==    by 0xA79041: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==29752== 
==29752== 640 bytes in 1 blocks are still reachable in loss record 21 of 30
==29752==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1175CE9: aligned_malloc (alloc.rs:95)
==29752==    by 0x1175CE9: alloc (alloc.rs:22)
==29752==    by 0x1175CE9: __rdl_alloc (alloc.rs:304)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA50695: alloc::sync::Arc<T>::new (sync.rs:324)
==29752==    by 0xA57E27: rayon_core::registry::Registry::new (registry.rs:238)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752== 
==29752== 640 bytes in 1 blocks are still reachable in loss record 22 of 30
==29752==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1175CE9: aligned_malloc (alloc.rs:95)
==29752==    by 0x1175CE9: alloc (alloc.rs:22)
==29752==    by 0x1175CE9: __rdl_alloc (alloc.rs:304)
==29752==    by 0xAC3C9B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xAC41E3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xAC3BFC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xABFE75: alloc::sync::Arc<T>::new (sync.rs:324)
==29752==    by 0xAC03B9: <crossbeam_epoch::collector::Collector as core::default::Default>::default (collector.rs:32)
==29752==    by 0xAC03E6: crossbeam_epoch::collector::Collector::new (collector.rs:40)
==29752==    by 0xABE8D9: __static_ref_initialize (default.rs:13)
==29752==    by 0xABE8D9: core::ops::function::FnOnce::call_once (function.rs:233)
==29752==    by 0xABE745: lazy_static::lazy::Lazy<T>::get::{{closure}} (inline_lazy.rs:31)
==29752==    by 0xABFA04: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752== 
==29752== 808 bytes in 1 blocks are still reachable in loss record 23 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xD7779B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xD778C3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xD776FC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xD85970: new<alloc::collections::btree::node::LeafNode<usize, wasmer_engine::trap::frame_info::ModuleInfoFrameInfo>> (boxed.rs:177)
==29752==    by 0xD85970: alloc::collections::btree::node::Root<K,V>::new_leaf (node.rs:157)
==29752==    by 0xD6C279: core::ops::function::FnOnce::call_once (function.rs:233)
==29752==    by 0xD789BC: core::option::Option<T>::get_or_insert_with (option.rs:853)
==29752==    by 0xD9E27E: alloc::collections::btree::map::BTreeMap<K,V>::ensure_root_is_owned (map.rs:2324)
==29752==    by 0xD9E651: alloc::collections::btree::map::BTreeMap<K,V>::entry (map.rs:1061)
==29752==    by 0xD9F4C9: alloc::collections::btree::map::BTreeMap<K,V>::insert (map.rs:829)
==29752==    by 0xD6739F: wasmer_engine::trap::frame_info::register (frame_info.rs:281)
==29752==    by 0x7FEA94: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (artifact.rs:275)
==29752== 
==29752== 1,152 bytes in 4 blocks are possibly lost in loss record 24 of 30
==29752==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==29752==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==29752==    by 0x5252227: allocate_stack (allocatestack.c:627)
==29752==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==29752==    by 0x11858FC: std::sys::unix::thread::Thread::new (thread.rs:66)
==29752==    by 0xA870FB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==29752==    by 0xA8832B: std::thread::Builder::spawn (mod.rs:386)
==29752==    by 0xA57148: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:101)
==29752==    by 0xA5822F: rayon_core::registry::Registry::new (registry.rs:260)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752== 
==29752== 1,520 bytes in 1 blocks are still reachable in loss record 25 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA66481: new<crossbeam_deque::deque::Block<rayon_core::job::JobRef>> (boxed.rs:177)
==29752==    by 0xA66481: <crossbeam_deque::deque::Injector<T> as core::default::Default>::default (deque.rs:1187)
==29752==    by 0xA64D0D: crossbeam_deque::deque::Injector<T>::new (deque.rs:1213)
==29752==    by 0xA57BF3: rayon_core::registry::Registry::new (registry.rs:242)
==29752==    by 0xA573BD: rayon_core::registry::global_registry::{{closure}} (registry.rs:169)
==29752==    by 0xA57749: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==29752==    by 0xA7AC13: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==29752==    by 0x1170929: std::sync::once::Once::call_inner (once.rs:416)
==29752==    by 0xA7AB90: std::sync::once::Once::call_once (once.rs:264)
==29752== 
==29752== 1,536 bytes in 4 blocks are still reachable in loss record 26 of 30
==29752==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0x1175CE9: aligned_malloc (alloc.rs:95)
==29752==    by 0x1175CE9: alloc (alloc.rs:22)
==29752==    by 0x1175CE9: __rdl_alloc (alloc.rs:304)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA504B5: alloc::sync::Arc<T>::new (sync.rs:324)
==29752==    by 0xA6483D: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:250)
==29752==    by 0xA586AE: rayon_core::registry::Registry::new::{{closure}} (registry.rs:229)
==29752==    by 0xA913BC: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA6200D: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752==    by 0xA92C13: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==29752== 
==29752== 2,008 bytes in 1 blocks are still reachable in loss record 27 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xAC3C9B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xAC41E3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xAC3BFC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xABC5B9: new<crossbeam_epoch::sync::queue::Node<crossbeam_epoch::internal::SealedBag>> (boxed.rs:177)
==29752==    by 0xABC5B9: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==29752==    by 0xABBF65: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==29752==    by 0xABBEB5: crossbeam_epoch::atomic::Owned<T>::new (atomic.rs:846)
==29752==    by 0xABC7BB: crossbeam_epoch::sync::queue::Queue<T>::new (queue.rs:50)
==29752==    by 0xAB7BB3: crossbeam_epoch::internal::Global::new (internal.rs:254)
==29752==    by 0xAC03AB: <crossbeam_epoch::collector::Collector as core::default::Default>::default (collector.rs:32)
==29752==    by 0xAC03E6: crossbeam_epoch::collector::Collector::new (collector.rs:40)
==29752==    by 0xABE8D9: __static_ref_initialize (default.rs:13)
==29752==    by 0xABE8D9: core::ops::function::FnOnce::call_once (function.rs:233)
==29752== 
==29752== 4,096 bytes in 4 blocks are still reachable in loss record 28 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA9A3ED: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==29752==    by 0xA9D660: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==29752==    by 0xA997DE: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==29752==    by 0xA74FEE: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==29752==    by 0xA63250: crossbeam_deque::deque::Buffer<T>::alloc (deque.rs:45)
==29752==    by 0xA64729: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:248)
==29752==    by 0xA586AE: rayon_core::registry::Registry::new::{{closure}} (registry.rs:229)
==29752==    by 0xA913BC: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==29752==    by 0xA6200D: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==29752== 
==29752== 6,080 bytes in 4 blocks are still reachable in loss record 29 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xA80BFB: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xA81A23: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xA80B5C: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xA66481: new<crossbeam_deque::deque::Block<rayon_core::job::JobRef>> (boxed.rs:177)
==29752==    by 0xA66481: <crossbeam_deque::deque::Injector<T> as core::default::Default>::default (deque.rs:1187)
==29752==    by 0xA64D0D: crossbeam_deque::deque::Injector<T>::new (deque.rs:1213)
==29752==    by 0xA5228C: rayon_core::job::JobFifo::new (job.rs:193)
==29752==    by 0xA5A64A: rayon_core::registry::main_loop (registry.rs:806)
==29752==    by 0xA56C7F: rayon_core::registry::ThreadBuilder::run (registry.rs:56)
==29752==    by 0xA57330: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn::{{closure}} (registry.rs:101)
==29752==    by 0xA8FBDF: std::sys_common::backtrace::__rust_begin_short_backtrace (backtrace.rs:130)
==29752==    by 0xA881B0: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}} (mod.rs:475)
==29752== 
==29752== 8,160 bytes in 4 blocks are still reachable in loss record 30 of 30
==29752==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29752==    by 0xAC3C9B: alloc::alloc::alloc (alloc.rs:80)
==29752==    by 0xAC41E3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==29752==    by 0xAC3BFC: alloc::alloc::exchange_malloc (alloc.rs:277)
==29752==    by 0xAB75E9: new<crossbeam_epoch::internal::Local> (boxed.rs:177)
==29752==    by 0xAB75E9: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==29752==    by 0xABBF05: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==29752==    by 0xABBE65: crossbeam_epoch::atomic::Owned<T>::new (atomic.rs:846)
==29752==    by 0xAB8452: crossbeam_epoch::internal::Local::register (internal.rs:392)
==29752==    by 0xAC040D: crossbeam_epoch::collector::Collector::register (collector.rs:45)
==29752==    by 0xABF1F3: crossbeam_epoch::default::HANDLE::__init (default.rs:18)
==29752==    by 0xABE8B8: core::ops::function::FnOnce::call_once (function.rs:233)
==29752==    by 0xAC438B: std::thread::local::lazy::LazyKeyInner<T>::initialize (local.rs:289)
==29752== 
==29752== LEAK SUMMARY:
==29752==    definitely lost: 0 bytes in 0 blocks
==29752==    indirectly lost: 0 bytes in 0 blocks
==29752==      possibly lost: 1,152 bytes in 4 blocks
==29752==    still reachable: 29,184 bytes in 86 blocks
==29752==         suppressed: 0 bytes in 0 blocks
==29752== 
==29752== For counts of detected and suppressed errors, rerun with: -v
==29752== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
